File: Interval List Intersections.cpp
class Solution {
public:
    vector<vector<int>> intervalIntersection(vector<vector<int>>& A, vector<vector<int>>& B) {
        vector<vector<int>>vec;
        for(int i=0,j=0;i<A.size() && j<B.size();)
        {
            if(!(A[i][0] > B[j][1] || A[i][1] < B[j][0]))
            {
                int l=max(A[i][0],B[j][0]);
                int r=min(A[i][1],B[j][1]);
                vec.push_back({l,r});
            }
            if(i<A.size()-1 && B[j][1]>=A[i+1][0])
            {
                i++;
                continue;
            }
            if(j<B.size()-1 && A[i][1]>=B[j+1][0])
            {
                j++;
                continue;
            }
            i++;
            j++;
        
        }
        return vec;
        
    }
};

File: Golang Solutions/Interval List Intersections.go
func intervalIntersection(A [][]int, B [][]int) [][]int {
    var vec [][]int
    var tmp []int
    for i,j:=0,0;i<len(A) && j<len(B); {
        if !(A[i][0] > B[j][1] || A[i][1] < B[j][0]){
            l:=int(math.Max(float64(A[i][0]),float64(B[j][0])))
            r:=int(math.Min(float64(A[i][1]),float64(B[j][1])))
            tmp=nil
            tmp=append(tmp,l)
            tmp=append(tmp,r)
            vec=append(vec,tmp)
        }
        if i<len(A)-1 && B[j][1]>=A[i+1][0]{
            i++
            continue
        }
        if j<len(B)-1 && A[i][1]>=B[j+1][0]{
            j++
            continue;
        }
        i++
        j++
        
    }
    return vec
}